packages/aws-cdk-lib/aws-ecr/lib/repository.ts

```diff
@@ -610,6 +631,55 @@ export interface RepositoryAttributes {
  * Define an ECR repository
  */
 export class Repository extends RepositoryBase {
+  /**
+   * Lookup an existing repository
+   */
+  public static fromLookup(scope: Construct, id: string, options: RepositoryLookupOptions): IRepository {
+    if (Token.isUnresolved(options.repositoryName) || Token.isUnresolved(options.repositoryArn)) {
+      throw new UnscopedValidationError('Cannot look up a repository with a tokenized name or ARN.');
+    }
+
+    if (!options.repositoryArn && !options.repositoryName) {
+      throw new UnscopedValidationError('At least one of `repositoryName` or `repositoryArn` must be provided.');
+    }
+
+    const identifier = options.repositoryName ??
+      (options.repositoryArn ? Arn.split(options.repositoryArn, ArnFormat.SLASH_RESOURCE_NAME).resourceName : undefined);
+
+    if (!identifier) {
+      throw new UnscopedValidationError('Could not determine repository identifier from provided options.');
+    }
+
+    const response: {[key: string]: any}[] = ContextProvider.getValue(scope, {
+      provider: cxschema.ContextProvider.CC_API_PROVIDER,
+      props: {
+        typeName: 'AWS::ECR::Repository',
+        exactIdentifier: identifier,
+        propertiesToReturn: ['Arn'],
+      } as cxschema.CcApiContextQuery,
+      dummyValue: [
+        {
+          Arn: Stack.of(scope).formatArn({
+            service: 'ecr',
+            region: 'us-east-1',
+            account: '123456789012',
+            resource: 'repository',
+            resourceName: 'DUMMY_ARN',
+          }),
+        },
+      ],
+    }).value;
+
+    const repository = response[0];
+    const lookupRepoName = Arn.split(repository.Arn, ArnFormat.SLASH_RESOURCE_NAME).resourceName;
+
+    if (!lookupRepoName) {
+      throw new UnscopedValidationError(`Unable to look up repository with name: ${options.repositoryArn} in account ${Aws.ACCOUNT_ID}`);
+    }
+
+    return this.fromRepositoryName(scope, id, lookupRepoName);
```

Comment:
We should use `fromRepositoryAttributes` for lookups, they tend to have a higher accuracy. In this case, I think we are might end up using the wrong region.

```suggestion
    return this. fromRepositoryAttributes(scope, id, {...});
```

Reply:
Thanks! I've updated my code to use `fromRepositoryAttributes()`.